### PR TITLE
feat(xrpc): register variable redis_cmd_line

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by tzssangglass** [comment]
base on #7040

File: docs/en/latest/apisix-variable.md [hunk 1]
```diff
@@ -39,5 +39,6 @@ List in alphabetical order:
 | route_name       | core    | name of `route`        |   |
 | service_id       | core    | id of `service`        |   |
 | service_name     | core    | name of `service`      |   |
+| redis_cmd_line   | Redis   | the content of Redis command |   |
```
Comment (kwanhur) [inline docs/en/latest/apisix-variable.md:42]: Better shorten name `redis_cmd` or `redis_command`?

`redis_cmd_line` seems it'll return multi commands compact into one line. It just return one total command content, not just a redis command like `hmset`, right?

File: apisix/stream/xrpc/protocols/redis/init.lua [hunk 2]
```diff
@@ -31,6 +31,15 @@ local ipairs = ipairs
 local tonumber = tonumber
 
 
+core.ctx.register_var("redis_cmd_line", function(ctx)
+    local curr_ctx = ctx.xrpc_session and ctx.xrpc_session._curr_ctx
+    if not curr_ctx then
+        core.log.warn("can't find current context")
+        return nil
+    end
+    return curr_ctx.cmd
```
Comment (spacewander) [inline apisix/stream/xrpc/protocols/redis/init.lua:40]: We should return cmd_line instead? https://github.com/apache/apisix/blob/1cdcc5caffc7ccf3825e3be43f62af8388b93551/apisix/stream/xrpc/protocols/redis/init.lua#L216

We can join the array with ` ` as this var is used in the log.

Comment (tzssangglass) [reply apisix/stream/xrpc/protocols/redis/init.lua:40]: > as this var is used in the log.

No, cmd_line is also used for filter, filter would be weird to write if it returned a spliced cmd_line.

cmd_line in array maybe like:

```
{ "hmget", "animals", "dog", "cat" },
```

cmd_line in string maybe like:

```
"hmget animals dog cat"
```

Comment (tzssangglass) [reply apisix/stream/xrpc/protocols/redis/init.lua]: Ok, I know, redis_cmd_line will only be used for logs, not filters.

File: docs/en/latest/apisix-variable.md (hunk 1 above)
Comment (tzssangglass) [reply docs/en/latest/apisix-variable.md:42]: yes, what it really means is command line.

File: apisix/stream/xrpc/protocols/redis/init.lua [hunk 3]
```diff
@@ -31,6 +31,14 @@ local ipairs = ipairs
 local tonumber = tonumber
 
 
+core.ctx.register_var("redis_cmd_line", function(ctx)
+    local cmd_line = ctx.cmd_line
+    if cmd_line and #cmd_line > 1 then
+        return core.table.concat(cmd_line, " ")
+    end
+    return ctx.cmd_line
```
Comment (spacewander) [inline apisix/stream/xrpc/protocols/redis/init.lua:39]: The ctx.cmd_line can be `["ping"]`?

Comment (tzssangglass) [reply apisix/stream/xrpc/protocols/redis/init.lua]: fixed

**Review by spacewander — APPROVED** [review]

File: t/xrpc/redis2.t [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+    if (!defined $block->extra_stream_config) {
+        my $stream_config = <<_EOC_;
+    server {
+        listen 8125 udp;
+        content_by_lua_block {
+            require("lib.mock_layer4").dogstatsd()
```
Comment (soulbird) [inline t/xrpc/redis2.t:55]: Better to use a new service, or rename `dogstatsd `.

**Review by soulbird — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**